docs/migrate-from-sidecar-to-ambient/migration.adoc

```diff
@@ -0,0 +1,1607 @@
+bookinfo                                         details-v1-584b5c5f6c-dhkjx                                 10.128.2.67   user-rhos-d-4-9b684-worker-0-4fhm6 None     HBONE
+bookinfo                                         productpage-v1-7b5749b8bf-96mmg                             10.128.2.68   user-rhos-d-4-9b684-worker-0-4fhm6 None     HBONE
+bookinfo                                         ratings-v1-cd589f595-wrjds                                  10.128.2.69   user-rhos-d-4-9b684-worker-0-4fhm6 None     HBONE
+bookinfo                                         reviews-v1-695599f9d5-58626                                 10.128.2.70   user-rhos-d-4-9b684-worker-0-4fhm6 None     HBONE
+bookinfo                                         reviews-v2-6bc5c6b6cf-4nqpq                                 10.128.2.71   user-rhos-d-4-9b684-worker-0-4fhm6 None     HBONE
+bookinfo                                         reviews-v3-5866bd5557-g86wx                                 10.128.2.72   user-rhos-d-4-9b684-worker-0-4fhm6 None     HBONE
+bookinfo                                         waypoint-748b7f45c6-vxlnd                                   10.128.2.73   user-rhos-d-4-9b684-worker-0-4fhm6 None     TCP
+----
+
+**Validate using Kiali:**
+After migration, use Kiali to visualize the service mesh. You should see traffic flowing through ztunnel and waypoints instead of sidecars. You will see the ztunnel as a node in the graph, and the waypoints as nodes for each namespace.
+
+[[troubleshooting]]
+== 5. Troubleshooting
+
+[[common-issues]]
+=== 5.1 Common Issues
+
+For comprehensive troubleshooting guidance, refer to:
+- https://istio.io/latest/docs/ambient/usage/troubleshoot-waypoint/[Istio Ambient Troubleshooting Documentation]
+- https://istio.io/latest/docs/ops/diagnostic-tools/[Istio Diagnostic Tools]
+
+==== Step-Specific Error Scenarios
+
+**Errors - Cluster Setup Issues:**
+
+*Issue: ZTunnel pods failing to start*
+[source,console]
+----
+# Check ZTunnel DaemonSet status
+kubectl get daemonset -n ztunnel
+kubectl describe daemonset -n ztunnel
+
+# Common causes and solutions:
+# 1. Node selector issues
+kubectl get nodes --show-labels
+# 2. Resource constraints
+kubectl describe nodes | grep -A 5 "Allocated resources"
+# 3. Security context issues
+kubectl logs -n ztunnel -l app=ztunnel --tail=50
+----
+
+**Errors - HBONE Support Issues:**
+
+*Issue: Pods fail to restart after enabling HBONE*
+[source,console]
+----
+# Check deployment rollout status
+kubectl rollout status deployment -n <NAMESPACE>
+
+# Check for image pull errors or resource limits
+kubectl describe pods -n <NAMESPACE> | grep -A 10 "Events:"
+
+# Verify sidecar proxy version supports HBONE
+kubectl exec -n <NAMESPACE> $(kubectl get pods -n <NAMESPACE> -l app=<SERVICE_NAME> -o jsonpath='{.items[0].metadata.name}') -c istio-proxy -- pilot-agent version
```

Comment:
I think we can remove this step as "pilot-agent version" does not really help in verifying that proxy version supports HBONE. It only lists the proxy version. The following step below should be sufficient to verify if HBONE is enabled.